envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyIngressRoutesFactory.kt

```diff
@@ -183,67 +158,13 @@ internal class EnvoyIngressRoutesFactory(
 
         val customHealthCheckRoute = customHealthCheckRoute(proxySettings)
 
-        if (!proxySettings.incoming.permissionsEnabled) {
-            return customHealthCheckRoute + allOpenIngressRoutes(localRouteAction)
-        }
-
-        val rolesByName = proxySettings.incoming.roles.associateBy { it.name.orEmpty() }
-
-        val applicationRoutes = proxySettings.incoming.endpoints
-            .flatMap { toRoutes(it, rolesByName, localRouteAction) }
-
-        return customHealthCheckRoute + listOfNotNull(
-            statusRoute(localRouteAction).takeIf { properties.routes.status.enabled }
-        ) + applicationRoutes + fallbackIngressRoute
-    }
-
-    private fun toRoutes(
-        endpoint: IncomingEndpoint,
-        roles: Map<String, Role>,
-        localRouteAction: RouteAction.Builder
-    ): List<Route> {
-        val routeMatch = RouteMatch.newBuilder()
-
-        when (endpoint.pathMatchingType) {
-            PathMatchingType.PATH -> routeMatch.path = endpoint.path
-            PathMatchingType.PATH_PREFIX -> routeMatch.prefix = endpoint.path
-        }
-
-        val clients = endpoint.clients
-            .flatMap { roles[it]?.clients ?: listOf(it) }
-            .distinct()
-
-        val methods = endpoint.methods
-            .takeIf { it.isNotEmpty() }
-            ?.map { HttpMethod.valueOf(it) }
-            ?: perMethodRetryPolicies.keys
-
-        return clients.flatMap { client ->
-            val routesForMethods = methods.map { method ->
-                val match = routeMatch.clone()
-                    .addHeaders(clientNameMatcher(client))
-                    .addHeaders(httpMethodMatcher(method))
-                Route.newBuilder()
-                    .setMatch(match)
-                    .setRoute(localClusterRouteActionWithRetryPolicy(method, localRouteAction))
-                    .build()
-            }
-            if (endpoint.methods.isEmpty()) {
-                val match = routeMatch.clone()
-                    .addHeaders(clientNameMatcher(client))
-                routesForMethods + Route.newBuilder()
-                    .setMatch(match)
-                    .setRoute(localRouteAction)
-                    .build()
-            } else routesForMethods
-        }
+        return customHealthCheckRoute +
+                listOfNotNull(
+                    statusRoute(localRouteAction).takeIf { properties.routes.status.enabled }
```

Comment:
statusRoute functionality is not ported to RBAC. We should:
* create ALLOW action for all for statuses (using `properties.routes.status.pathPrefix`)
* remove statusRoute from this class, it is not needed